SPECS/glusterfs/glusterfs.spec

```diff
@@ -34,463 +45,666 @@
 # provides are filtered
 %global __provides_exclude_from ^%{_libdir}/glusterfs/%{version}/.*$
+%endif
+%global bashcompdir %(pkg-config --variable=completionsdir bash-completion 2>/dev/null)
+%if "%{bashcompdir}" == ""
+%global bashcompdir ${sysconfdir}/bash_completion.d
+%endif
+%if ( 0%{?mariner} && 0%{?mariner} >= 3 )
+# Skip regression-tests because they previously were excluded in
+# CBL-Mariner glusterfs and the new version introduces a dependency
+# (dbench)
+%global _without_regression_tests true
+# Explicitly require rpcgen
+%global _require_rpcgen true
+%endif
 ##-----------------------------------------------------------------------------
 ## All package definitions should be placed here and keep them sorted
 ##
 Summary:        Distributed File System
 Name:           glusterfs
-Version:        7.9
-Release:        5%{?dist}
+Version:        11.1
+Release:        1%{?dist}
 License:        GPLv2 OR LGPLv3+
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 URL:            https://docs.gluster.org/
-Source0:        https://download.gluster.org/pub/gluster/%{name}/7/%{version}/%{name}-%{version}.tar.gz
+Source0:        https://download.gluster.org/pub/gluster/%{name}/11/%{version}/%{name}-%{version}.tar.gz
 Source1:        glusterd.sysconfig
 Source2:        glusterfsd.sysconfig
 Source7:        glusterfsd.service
 Source8:        glusterfsd.init
-Patch0001:      0001-rpc-rpc-lib-src-Makefile.am.patch
-
-BuildRequires:  bison
-BuildRequires:  flex
-BuildRequires:  gcc
-BuildRequires:  libacl-devel
-BuildRequires:  libaio-devel
-BuildRequires:  libtool
-BuildRequires:  libuuid-devel
-BuildRequires:  libxml2-devel
-BuildRequires:  make
-BuildRequires:  ncurses-devel
-BuildRequires:  openssl-devel
-BuildRequires:  python%{_pythonver}-devel
-BuildRequires:  readline-devel
-BuildRequires:  rpcgen
-BuildRequires:  systemd
-BuildRequires:  userspace-rcu-devel >= 0.7
+%if ( 0%{?mariner} && 0%{?mariner} >= 3 )
+# Patch created for Mariner to fix build break related to
+# including eventtypes.h
+Patch001:       include-eventtypes-always.patch
+%endif
 
-%if ( 0%{?_with_firewalld:1} )
-BuildRequires:  firewalld
+BuildRoot:        %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
+
+Requires(pre):    shadow-utils
+%if ( 0%{?_with_systemd:1} )
+BuildRequires:    systemd
 %endif
-%if 0%{?_with_asan:1}
-BuildRequires:  libasan
+
+%if ( 0%{!?_without_tcmalloc:1} )
+Requires:         gperftools-libs%{?_isa}
 %endif
-%if ( 0%{!?_without_georeplication:1} )
-BuildRequires:  libattr-devel
+
+Requires:         libglusterfs0%{?_isa} = %{version}-%{release}
+Requires:         libgfrpc0%{?_isa} = %{version}-%{release}
+Requires:         libgfxdr0%{?_isa} = %{version}-%{release}
+%if ( 0%{?_with_systemd:1} )
+%{?systemd_requires}
+%endif
+%if 0%{?_with_asan:1} && !( 0%{?rhel} && 0%{?rhel} < 7 )
+BuildRequires:    libasan
+%endif
+%if 0%{?_with_tsan:1} && !( 0%{?rhel} && 0%{?rhel} < 7 )
+BuildRequires:    libtsan
```

Comment:
Please keep the order of `BuildRequires` coming first in alhpabetical order, then `Requires`, `Provides`, and `Obsoletes`. If `Provides` and `Obsoletes` form "pairs", it's also fine to keep these pairs together.

Reply:
Actually, since this is a new set of packages for 3.0, all `Provides`-`Obsolete` "pairs" like `%{name}-common` for instance can be removed completely. There's no older package to obsolete.

`Provides` without pairs should probably still remain, though.

Reply:
this is the upstream spec.  if we change it, the next upgrade will be harder.

Reply:
have grouped like things and alphabetized them.  left provides-obselete pairs because they existed previously.